tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_container.ts

```diff
@@ -60,17 +64,34 @@ const INPUT_CHANGE_DEBOUNCE_INTERVAL_MS = 500;
 })
 export class RegexEditDialogContainer {
   private readonly experimentIds: string[];
+  private readonly expNameByExpId: Record<string, string>;
   private readonly runIdToEid$: Observable<Record<string, string>>;
   private readonly allRuns$: Observable<Run[]>;
   private readonly tentativeRegexString$: Subject<string> =
     new Subject<string>();
+  private readonly tentativeRegexType$: Subject<GroupByKey> =
```

Comment:
It wasn't immediately clear why the tentative values are needed. Could we add a short comment saying that these changes are local to the component and are used before pressing save?

Reply:
In this case I wanted to mirror groupByRegexString, since they both have the same behavior here. I will write a comment explaining that the tentative variables are used as a callback for a change in either one of them.

Reply:
The comment clearly describes "what" these do but not "why". Could we emphasize that these allow updating the ngrx state only on save button click? (and not on every dropdown change)